modules/apps/commerce/commerce-payment-service/src/main/java/com/liferay/commerce/payment/internal/engine/CommercePaymentEngineImpl.java

```diff
@@ -540,8 +544,16 @@ private List<CommercePaymentMethod> _getCommercePaymentMethodsList(
 				_commercePaymentMethodRegistry.getCommercePaymentMethod(
 					commercePaymentMethodGroupRel.getEngineKey());
 
-			if (subscriptionOrder &&
-				!commercePaymentMethod.isProcessRecurringEnabled()) {
+			boolean hasPermission = permissionChecker.hasPermission(
+				commercePaymentMethodGroupRel.getGroupId(),
+				CommercePaymentMethodGroupRel.class.getName(),
+				commercePaymentMethodGroupRel.
+					getCommercePaymentMethodGroupRelId(),
+				ActionKeys.VIEW);
+
+			if (!hasPermission ||
+				(subscriptionOrder &&
+				 !commercePaymentMethod.isProcessRecurringEnabled())) {
```

Comment:
```suggestion
			if (!permissionChecker.hasPermission(
					commercePaymentMethodGroupRel.getGroupId(),
					CommercePaymentMethodGroupRel.class.getName(),
					commercePaymentMethodGroupRel.
						getCommercePaymentMethodGroupRelId(),
					ActionKeys.VIEW) ||
				(subscriptionOrder &&
				 !commercePaymentMethod.isProcessRecurringEnabled())) {
```

Reply:
@brianikim just to be on the safe side on getting this passed, we can remove the variable hasPermission since its only being used in one place.